filecoin-proofs/src/api/post_adapter.rs

```diff
@@ -0,0 +1,626 @@
+    let faults_len = dynamic.faults.len();
+    let commrs_len = dynamic.comm_rs.len();
+    let proofs_len = dynamic.proofs.len();
+    let replicas_c = (commrs_len as f64 / POST_SECTORS_COUNT as f64)
+        .ceil()
+        .round() as usize;
+    let faults_max = dynamic.faults.iter().max();
+
+    if faults_len > commrs_len {
+        return Err(format_err!(
+            "LEN(faults) must <= LEN(comm_rs): {:?}, {:?}",
+            faults_len,
+            commrs_len
+        ));
+    }
+
+    if proofs_len != replicas_c {
+        return Err(format_err!(
+            "LEN(proofs) must == CEIL(LEN(comm_rs)/2): {:?}, {:?}",
+            proofs_len,
+            replicas_c
+        ));
+    }
+
+    if faults_max
+        .map(|m| *m > commrs_len as u64 - 1)
+        .unwrap_or(false)
+    {
+        return Err(format_err!(
+            "MAX(faults) must <= LEN(comm_rs)-1: {:?}, {:?}",
+            faults_max,
+            commrs_len - 1
+        ));
+    }
+
+    let mut fixed: Vec<VerifyPoStFixedSectorsCountInput> = vec![];
+
+    let chunks = dynamic.comm_rs.chunks_exact(POST_SECTORS_COUNT);
+
+    let remainder = chunks.remainder();
+
+    // each POST_SECTORS_COUNT grouping of commitments maps to a single fixed
+    // sectors-count PoSt verification call
+    for (i, chunk) in chunks.enumerate() {
+        let mut comm_rs: [Commitment; POST_SECTORS_COUNT] = [[0; 32]; POST_SECTORS_COUNT];
+        for (i, comm_r) in chunk.iter().enumerate() {
+            comm_rs[i] = *comm_r;
+        }
+
+        fixed.push(VerifyPoStFixedSectorsCountInput {
+            sector_bytes: dynamic.sector_bytes,
+            comm_rs,
+            challenge_seed: dynamic.challenge_seed,
+            proof: dynamic.proofs[i],
+            faults: Vec::new(),
+        })
+    }
+
+    // If we receive a number of comm_rs which does not divide evenly by
+    // POST_SECTORS_COUNT, form a grouping of comm_rs which does divide
+    // evenly by POST_SECTORS_COUNT by duplicating the last comm_r in the
+    // group until LEN(comm_rs) == POST_SECTORS_COUNT.
+    if !remainder.is_empty() {
+        let mut comm_rs: [Commitment; POST_SECTORS_COUNT] = Default::default();
+
+        let vs = repeat(remainder[remainder.len() - 1]);
+        let rs = remainder.iter().cloned();
+        let it = rs.chain(vs).take(POST_SECTORS_COUNT).enumerate();
+
+        for (i, comm_r) in it {
+            comm_rs[i] = comm_r;
+        }
+
+        fixed.push(VerifyPoStFixedSectorsCountInput {
+            sector_bytes: dynamic.sector_bytes,
+            comm_rs,
+            challenge_seed: dynamic.challenge_seed,
+            proof: dynamic.proofs[proofs_len - 1],
+            faults: Vec::new(),
+        })
+    }
+
+    // Map the dynamic sectors count indices to the fixed sectors count indices.
+    for fault in dynamic.faults {
+        let i = (fault as f64 / POST_SECTORS_COUNT as f64).floor();
+        fixed[i as usize]
```

Comment:
As in `generate_`, I can't thoroughly follow this logic because I don't know the pattern of the `fault` values (what confuses me is that we seem to be mixing `fault` values with the index we're computing to store it).

Reply:
Good call. I added a little example in the comments, which I'll repaste here:

```

    // For example:
    //
    // dynamic = {faults=[0, 1, 5, 6] and POST_SECTORS_COUNT = 2
    // fixed = [{faults=[0, 1]}, {faults=[]}, {faults=[1]}, {faults=[0]}]
```